tests/unit/serialization/custom_global_serialization_test.py

```diff
@@ -128,3 +140,34 @@ def test_double_register_custom_serializer(self):
 
         with self.assertRaises(ValueError):
             service._registry.safe_register_serializer(TheOtherCustomSerializer, CustomClass)
+
+    def test_serializing_class_instances(self):
+        config = Config()
+        config.custom_serializers = {CustomClass: CustomSerializer}
+        service = SerializationServiceV1(config)
+
+        data1 = service.to_data(CustomClass)
+        deserialized1 = service.to_object(data1)
+
+        self.assertEqual(CustomClass, deserialized1)
+
+        data2 = service.to_data(TheOtherCustomClass)
+        deserialized2 = service.to_object(data2)
+
+        self.assertEqual(TheOtherCustomClass, deserialized2)
+
+    def test_serializing_class_and_object_instances(self):
```

Comment:
```suggestion
    def test_serializing_child_class_instances_with_super_class_serializer(self):
```

Reply:
How do we determine the test names?

Reply:
Well, we try to be as precise as we can so that we can know what the test does in the future by reading the test name. The former name didn't have the necessary info IMO